RTSS-92 Front End For Payment Method

AdongoJr2 opened the pull request.

AdongoJr2 committed:
RTSS-91 Implement Payment Method CRUD APIs

abishai53 commented on src/lib/api/payment-option/payment-option.service.ts:
```diff
@@ -0,0 +1,14 @@
+@Injectable({ providedIn: 'root' })
+export class PaymentOptionService extends BaseService<PaymentOption, PaymentOptionState> {
+  constructor(protected override readonly store: PaymentOptionStore,
+    protected override readonly query: PaymentOptionQuery) {
```
tabbing is off

abishai53 commented on src/lib/components/private/payment-option/payment-option-form/payment-option-form.component.ts:
```diff
@@ -0,0 +1,58 @@
+  private readonly paymentOptionService = inject(PaymentOptionService)
+  private readonly formBuilder = inject(FormBuilder)
+
+  readonly paymentOptionForm = computed(() => this.formBuilder.nonNullable.group({
```
nonNullable? the description is nullable

AdongoJr2 replied:
I think the `nonNullable` is for the formBuilder and not the individual fields. Adding a payment method without providing the `description` sends the value as `null` to the backend

abishai53 commented on src/lib/components/private/payment-option/payment-option.component.html:
```diff
@@ -0,0 +1,57 @@
+    <h3 class="font-semibold">Payment Options</h3>
+    <hr />
+
+    <p-table [loading]="(loading$ | async) && !addingIsActive()" selectionMode="single" rowExpandMode="single"
```
each attribute gets its own line

abishai53 commented on src/lib/components/private/payment-option/payment-option.component.html:
```diff
@@ -0,0 +1,57 @@
+          <td>{{paymentOption?.createdBy | nullSafe}}</td>
+          <td>{{paymentOption?.usageCount | nullishToZero}}</td>
+          <td>
+            <p-button [icon]="expanded ? 'pi pi-times' : 'pi pi-pen-to-square'" rounded severity="secondary"
```
each attribute gets a line

abishai53 commented on src/lib/components/private/payment-option/payment-option.component.ts:
```diff
@@ -0,0 +1,54 @@
+    this.subscriptions.add(this.selectPaymentOptionOnInitialLoad())
+  }
+
+  private selectPaymentOptionOnInitialLoad(): Subscription {
```
This grid won't need to selection on load. Actually, the rows don't need to be selectable at all. Just the edit button on the right is enough

AdongoJr2 committed:
Make code refactors based on PR comments:

- Made each attribute be in its own line in .html files
- Fixed tabbing that that is off
- Remove selection functionality from payment options table

abishai53 commented on src/lib/components/private/payment-option/payment-option.component.ts:
```diff
@@ -0,0 +1,42 @@
+  readonly addingIsActive = signal(false)
+  readonly rowIsExpanded = signal<boolean>(false)
+
+  override ngOnInit() {
```
not needed since all it does is call super

AdongoJr2 committed:
Remove unnecessary `ngOnInit` method in `PaymentOptionComponent`

Additional code refactors:
- Also removed unused import statements in `PaymentOptionComponent`

Commit:
Merge remote-tracking branch 'origin/develop' into RTSS-92-Front-End-for-payment-method

Commit:
RTSS-92 added filters on payment options

Commit:
RTSS-92 lint fixes
